src/Artifacts.jl

```
@@ -0,0 +1,589 @@
+            elseif any((unpack_platform(x, name, artifact_toml) for x in meta) .== Ref(platform))
+                error("Mapping for '$name'/$(triplet(platform)) within $(artifact_toml) already exists!")
+            end
+        end
+    else
+        artifact_dict = Dict()
+    end
+
+    # Otherwise, the new piece of data we're going to write out is this dict:
+    meta = Dict{String,Any}(
+        "git-tree-sha1" => bytes2hex(hash.bytes),
+    )
+
+    # If we're set to be lazy, then lazy we shall be
+    if lazy
+        meta["lazy"] = true
+    end
+
+    # Integrate download info, if it is given.  We represent the download info as a
+    # vector of dicts, each with its own `url` and `sha256`, since different tarballs can
+    # expand to the same tree hash.
+    if download_info != nothing
+        meta["download"] = [
+            Dict("url" => dl[1],
+                 "sha256" => dl[2],
+            ) for dl in download_info
+        ]
+    end
+
+    if platform == nothing
+        artifact_dict[name] = meta
+    else
+        # Add platform-specific keys to our `meta` dict
+        pack_platform!(meta, platform)
+
+        # Insert this entry into the list of artifacts
+        if !haskey(artifact_dict, name)
+            artifact_dict[name] = [meta]
+        else
+            # Delete any entries that contain identical platforms
+            artifact_dict[name] = filter(
+                x -> unpack_platform(x, name, artifact_toml) != platform,
+                artifact_dict[name]
+            )
+            push!(artifact_dict[name], meta)
+        end
+    end
+
+    # Spit it out onto disk
+    open(artifact_toml, "w") do io
+        TOML.print(io, artifact_dict, sorted=true)
+    end
+    return
+end
+
+
+"""
+    unbind_artifact(name::String, artifact_toml::String; platform = nothing)
+
+Unbind the given `name` from an `Artifact.toml` file.  Silently fails if no such binding
```

What does "silently fails" mean? Is that the same as "do nothing"?

---

Yes.